memebot/hooks.py

```diff
@@ -0,0 +1,45 @@
+import abc
+import os
+import logging
+import re
+import discord
+from typing import Set
+
+
+log = logging.getLogger('memebot')
+
+
+class Hook(abc.ABC):
```

Comment:
Why do we need a hook when we can just look at the messages?

Reply:
Didn't want to add this to `memebot.py` cuz already has enough code as it is. Also this should make it easier to add easter eggs whenever someone says something.